### PR TITLE
🌱 Improve v1beta2 condition ordering

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes-sigs/cluster-api/issues/11105

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold 
I forgot some conditions

File: util/conditions/v1beta2/sort.go [hunk 1]
```diff
@@ -38,26 +46,90 @@ func defaultSortLessFunc(i, j metav1.Condition) bool {
...
+// | EtcdPodHealthy                 |         |     |    |    |    | x       |
+// | EtcdMemberHealthy              |         |     |    |    |    | x       |
+// | HealthCheckSucceeded           |         |     |    |    |    | x       |
+// | OwnerRemediated                |         |     |    |    |    | x       |
+// | -- Operations --               |         |     |    |    |    |         |
+// | TopologyReconciled             | x       |     |    |    |    |         |
+// | Remediating                    | x       | x   | x  | x  | x  |         |
+// | ScalingDown                    | x       | x   | x  | x  | x  |         |
+// | ScalingUp                      | x       | x   | x  | x  | x  |         |
+// | UpToDate                       |         |     |    |    |    | x       |
```
Comment (fabriziopandini) [inline util/conditions/v1beta2/sort.go:83]: I will move UpToDate right after Ready, so for the machine, we have Available, Ready, UpToDate at the beginning.
Also, UpToDate is not really an operaiton

File: util/conditions/v1beta2/sort.go [hunk 2]
```diff
@@ -38,26 +46,90 @@ func defaultSortLessFunc(i, j metav1.Condition) bool {
...
-		return li < lj
-	}
-
 	return i.Type < j.Type
 }
 
-var first = map[string]int{
-	clusterv1.AvailableV1Beta2Condition: 0,
-	clusterv1.ReadyV1Beta2Condition:     1,
+// The order array below leads to the following condition ordering:
```
Comment (fabriziopandini) [inline util/conditions/v1beta2/sort.go:52]: Nice table!!

File: util/conditions/v1beta2/sort.go [hunk 3]
```diff
@@ -38,26 +46,90 @@ func defaultSortLessFunc(i, j metav1.Condition) bool {
...
+// | Initialized                    |         | x   |    |    |    |         |
+// | EtcdClusterHealthy             |         | x   |    |    |    |         |
+// | ControlPlaneComponentsHealthy  |         | x   |    |    |    |         |
+// | NodeHealthy                    |         |     |    |    |    | x       |
+// | NodeReady                      |         |     |    |    |    | x       |
+// | APIServerPodHealthy            |         |     |    |    |    | x       |
+// | ControllerManagerPodHealthy    |         |     |    |    |    | x       |
+// | SchedulerPodHealthy            |         |     |    |    |    | x       |
+// | EtcdPodHealthy                 |         |     |    |    |    | x       |
+// | EtcdMemberHealthy              |         |     |    |    |    | x       |
```
Comment (fabriziopandini) [inline util/conditions/v1beta2/sort.go:75]: What about moving Etcd conditions right before API server (if etcd is broken, it should be fixed before the API server)

File: util/conditions/v1beta2/sort.go (hunk 1 above)
Comment (sbueringer) [reply util/conditions/v1beta2/sort.go:83]: It's more part of an operation than an availability condition. But I think it's okay to have it next to ready.

So we have Ready, UpToDate and on others objects MachinesReady, MachinesUpToDate

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**